test-app/startup_arm.c

```diff
@@ -29,15 +29,18 @@ extern unsigned int _end_bss;
 extern unsigned int _end_stack;
 extern unsigned int _start_heap;
 
-static int zeroed_variable_in_bss;
-static int initialized_variable_in_data = 42;
-
+#ifdef STM32F
```

Comment:
I am not seeing `STM32F`? Perhaps you meant the new `STM32` one in test-app/MakeFile:53? Perhaps consider renaming this, since it is for all ARM archs?